### ISSUE TITLE
Support for C:\Users\<Account>\AppData\<AppName> pattern

### ISSUE THREAD

**Issue by darmbrust** [harawata/appdirs#28]
You have many patterns, but too many parameters aren't optional, and I don't want to use version and appAuthor.  

Especially version may make no sense if an app wants to find the existing data after an upgrade, and continue using it....

**Comment by darmbrust** [comment]
Right now, one must mis-use the API like this, to get the pattern that I'm looking for..:
AppDirsFactory.getInstance().getUserDataDir(null, "appNameInVersionSlot", null)

**Comment by harawata** [comment]
Thank you for the idea, @darmbrust !
I'll make it configurable. Give me some time, please.

**Comment by harawata** [comment]
Hi @darmbrust ,

A new `getInstance()` method which takes an instance of `AppDirsOptions` is added to `AppDirsFactory`.
To ignore appAuthor on Windows, enable `ignoreAppAuthorOnWindows` switch like  this:

```java
AppDirsOptions options = new AppDirsOptions().ignoreAppAuthorOnWindows();
AppDirs appDirs = AppDirsFactory.getInstance(options);
System.out.println(appDirs.getUserDataDir("myapp", null, null));
```

On Windows 10 (development environment), the output is:

```
C:\Users\User\AppData\Local\myapp
```

Note that when this option is enabled, the appAuthor parameter is ignored even if you specified a non-null value.

Let me know if this does not meet your requirement or there is a better approach!

---

I would appreciate if you could you try the latest [1.0.4-SNAPSHOT](https://oss.sonatype.org/content/repositories/snapshots/net/harawata/appdirs/).

If you use Maven, add the snapshot repository to the pom.xml...

```xml
<project>
...
  <repositories>
    <repository>
      <id>sonatype-snapshot</id>
      <name>Sonatype OSS Snapshot Repository</name>
      <url>http://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
  </repositories>
...
</project>
```
...and specify `1.0.4-SNAPSHOT` as appdirs version.

```xml
<dependency>
  <groupId>net.harawata</groupId>
  <artifactId>appdirs</artifactId>
  <version>1.0.4-SNAPSHOT</version>
</dependency>
```

If you're busy, please let me know and I'll just release it and pray. :D

**Comment by jmp** [comment]
I have the same use case as darmbrust (i.e. I have no use for both app author or version).

1.0.4-SNAPSHOT solves the problem. But I wonder if the API could be simpler, for example `null` would mean that argument would be ignored:
```java
AppDirs appDirs = AppDirsFactory.getInstance();
System.out.println(
    appDirs.getUserConfigDir(
        "myapp",
        null, // appVersion ignored
        null  // appAuthor ignored
    )
); // -> C:\Users\User\AppData\Local\myapp
```
or even simpler:
```java
AppDirs appDirs = AppDirsFactory.getInstance();
System.out.println(
    appDirs.getUserConfigDir("myapp")
); // -> C:\Users\User\AppData\Local\myapp
```
But maybe that (at least the first option) would break the existing API.

**Comment by harawata** [comment]
Thanks for the comment, @jmp !

The idea was to keep the backward compatibility.
Please give me a few days and I'll see if there is an alternative approach.

**Comment by harawata** [comment]
I have committed the change #32 and it now works as @jmp 's first example. i.e.

```java
AppDirs appDirs = AppDirsFactory.getInstance();
System.out.println(
    appDirs.getUserConfigDir(
        "myapp",
        null, // appVersion ignored
        null  // appAuthor ignored
    )
); // -> C:\Users\xxx\AppData\Local\myapp
```
Regarding the fewer-argument versions of the signatures (`appName` and `appVersion` only, `appName` only), let me revisit it later as I am a little bit busy right now.

The next version will be 1.1.0, so if you want to verify the new behavior, please use 1.1.0-SNAPSHOT and let me know if there is any issue.

Thank you both for the ideas!

**Comment by harawata** [comment]
Released [appdirs-1.1.0](https://github.com/harawata/appdirs/releases/tag/appdirs-1.1.0).
It's also available on Maven central repository.
